testsuite_overview.md

```diff
@@ -0,0 +1,310 @@
+Rest.li Client Test Suite
+=========================
+
+This directory contains a language-independent Rest.li client test suite.
+It is designed to be shared across languages, and provides test data and testing guidelines that can be followed when 
+testing a particular language implementation of the Rest.li client.
+
+This project also contains a Java TestNG suite that uses the test suite's data and test guidelines. 
+This example demonstrates how to use the test suite for a specific language implementation of the Rest.li client. 
+
+
+Motivation
+----
+This test suite allows us to compare implementations of the Rest.li client across
+languages.
+It will help us achieve consistently high quality implementations of the Rest.li client in multiple
+languages by:
+
+* Reducing the work required to add a new Rest.li client implementation
+* Providing a common ground where test cases can be shared amongst multiple Rest.li client implementations
+* "Certifying" Rest.li client implementations with a quantifiable measure of quality
+* Making it easier to make compatible Rest.li implementations, so a Rest.li client in one language can be used with a Rest.li server in another language. 
+
+To this effect, we've added test suite support for Python and Java, and used the test suite to create this
+[Compatibility Matrix](testsuite_compatibility_matrix.md), which compares the features supported by the implementations.
+
+Getting Started
+----
+This documentation assumes a background in Rest.li. Implementers are encouraged to explore the
+test data and reference documentation in tandem to learn the details of Rest.li.
+
+* .pdscs - https://github.com/linkedin/rest.li/wiki/DATA-Data-Schema-and-Templates
+* .restspec.json - https://github.com/linkedin/rest.li/wiki/Rest.li-.restspec.json-Format
+* wire protocol - https://github.com/linkedin/rest.li/wiki/Rest.li-Protocol
+
+There are two main ways of interacting with this test suite: 
+
+1. Expanding the language-independent test suite by testing new features or cases
+2. Adding test suite support for a new language to test a Rest.li implementation
+
+These distinct ways of using the test suite point to an important distinction between language-independent
+and language-specific components within this project. The test suite consists of language-independent data and test guidelines, meant to be
+used for standardizing cross-platform testing. Yet, because it's language independent, the test suite cannot immediately be used to test a
+specific language implementation of the Rest.li client. As a user, you must first add support for the test suite in your
+desired language. That is why we refer to a general "test suite", and to a "Java TestNG test suite" 
```

Comment:
Throughout the documentation, we use "Rest.li TestSuite Specification" for its language-independent component, and use "Java TestNG Tests" or "Python pytest Tests" for its language-dependent components.